cmd/drone-docker-ecr/main.go

```diff
@@ -11,46 +11,40 @@ import (
 )
 
 const (
-	DOCKER_USER    = "AWS"
-	DEFAULT_REGION = "us-east-1"
+	DockerUser    = "AWS"
+	DefaultRegion = "us-east-1"
 )
 
 func main() {
 	var registryIds []*string
-
-	ecrRegion := DEFAULT_REGION
-
-	if getenv("ECR_REGION") != "" {
-		os.Setenv("PLUGIN_REGION", getenv("ECR_REGION"))
+	var (
+		ecrRegion  = getenv("ECR_REGION", "PLUGIN_REGION")
+		accessKey  = getenv("ECR_ACCESS_KEY", "PLUGIN_ACCESS_KEY")
+		secretKey  = getenv("ECR_SECRET_KEY", "PLUGIN_SECRET_KEY")
+		registries = getenv("ECR_REGISTRY_IDS", "PLUGIN_REGISTRY_IDS")
+	)
+
+	if ecrRegion == "" {
+		ecrRegion = DefaultRegion
 	}
 
-	if accessKey := getenv("ECR_ACCESS_KEY"); accessKey != "" {
-		os.Setenv("PLUGIN_ACCESS_KEY", accessKey)
-	}
-
-	if secretKey := getenv("ECR_SECRET_KEY"); secretKey != "" {
-		os.Setenv("PLUGIN_SECRET_KEY", secretKey)
-	}
-
-	if accessKey := getenv("PLUGIN_ACCESS_KEY"); accessKey != "" {
+	if accessKey != "" && secretKey != "" {
 		os.Setenv("AWS_ACCESS_KEY_ID", accessKey)
```

Comment:
This should also be moved to the top getenv call

Reply:
I don't fully understand, `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` are read by AWS SDK http://docs.aws.amazon.com/sdk-for-go/api/aws/session/#hdr-Environment_Variables

So if we expect them to only come as environment variables e.g. as secrets, then this code has to be removed.
```		
accessKey  = getenv("ECR_ACCESS_KEY", "PLUGIN_ACCESS_KEY")
secretKey  = getenv("ECR_SECRET_KEY", "PLUGIN_SECRET_KEY")
```

If that the case, it must be documented so user knows how to call the secrets i.e. `aws_access_key_id` and `aws_secret_access_key`. In addition it won't be possible to use `PLUGIN_ACCESS_KEY` and `PLUGIN_SECRET_KEY`

Reply:
Anyhow I will read if it's possible to pass both access and secret keys as strings programatically, which would solve the problem.

Reply:
Sorry, was reading it wrong, I think setting the env variables should be fine

Reply:
so is it ok as it's now? or would it be better using credentials? - which would be something like: https://gist.github.com/donkeysharp/1949f93e44011c5df83ad4112879293b (I didn't tested yet)